src/middlewares/authMiddleware.js

```diff
@@ -0,0 +1,20 @@
+import { jwtService } from '../services/jwt.service.js';
+
+export const authMiddleware = (req, res, next) => {
+  const authorization = req.headers['authorization'] || ' ';
+  const [, token] = authorization.split(' ');
+
+  if (!authorization || !token) {
+    res.sendStatus(401);
+    return;
+  }
+
+  const userData = jwtService.verify(token);
+
+  if (!userData) {
+    res.sendStatus(401);
+    return;
+  }
+
+  next();
```

Comment:
Consider attaching `userData` to the `req` object (e.g., `req.user = userData;`) before calling `next()`. This allows subsequent middleware or route handlers to access the authenticated user's data.